resources/lib/functions.py

```diff
@@ -1112,6 +1132,208 @@ def checkService():
         sys.exit()
 
 
+def search(handle, params):
+    log.info('search Called: ' + str(params))
+    item_type = params.get('item_type')
+    if not item_type:
+        return
+    kb = xbmc.Keyboard()
+    if item_type.lower() == 'movie':
+        heading_type = i18n('movies')
+    elif item_type.lower() == 'series':
+        heading_type = i18n('series')
+    elif item_type.lower() == 'episode':
+        heading_type = i18n('episodes')
+    else:
+        heading_type = item_type
+    kb.setHeading(heading_type.capitalize() + ' ' + i18n('search').lower())
+    kb.doModal()
+    if kb.isConfirmed():
+        user_input = kb.getText().strip()
+        if user_input:
+            xbmcplugin.endOfDirectory(handle, cacheToDisc=False)
+            user_input = urllib.quote(user_input)
+            xbmc.executebuiltin('Container.Update(plugin://plugin.video.embycon/?mode=SEARCH_RESULTS&query={user_input}&item_type={item_type}&index=0)'
+                                .format(user_input=user_input, item_type=item_type))  # redirect for results to avoid page refreshing issues
+        else:
+            return
+    else:
+        return
+
+
+def searchResults(params):
+    log.info('searchResults Called: ' + str(params))
+
+    handle = int(sys.argv[1])
+    query = params.get('query')
+    item_type = params.get('item_type')
+    if (not item_type) or (not query):
+        return
+
+    index = params.get('index', 0)
+    if not index.isdigit():
+        index = 0
+
+    index = int(index)
+    limit = 50
+
+    settings = xbmcaddon.Addon(id='plugin.video.embycon')
+    port = settings.getSetting('port')
+    host = settings.getSetting('ipaddress')
+    server = host + ':' + port
+    userid = downloadUtils.getUserId()
+    detailsString = getDetailsString()
+
+    content_url = ('http://' + server +
+                  '/emby/Search/Hints?searchTerm=' + query +
+                  '&IncludeItemTypes=' + item_type +
+                  '&UserId=' + userid +
+                  '&StartIndex=' + str(index) +
+                  '&Limit=' + str(limit) +
+                  '&IncludePeople=false&IncludeMedia=true&IncludeGenres=false&IncludeStudios=false&IncludeArtists=false')
+
+    if item_type.lower() == 'movie':
+        xbmcplugin.setContent(int(sys.argv[1]), 'movies')
+        media_type = 'movie'
+    elif item_type.lower() == 'series':
+        xbmcplugin.setContent(int(sys.argv[1]), 'tvshows')
+        media_type = 'tvshow'
+    elif item_type.lower() == 'episode':
+        xbmcplugin.setContent(int(sys.argv[1]), 'episodes')
+        media_type = 'episode'
+    else:
+        xbmcplugin.setContent(int(sys.argv[1]), 'videos')
+        media_type = 'video'
+
+    json_data = downloadUtils.downloadUrl(content_url, suppress=False, popup=1)
```

Comment:
SearchHints returns different item objects, didn't work with getContent like the rest does.